File: J/771-jewels-and-stones/main.go
package main

import (
	"fmt"
)

func numJewelsInStones(J string, S string) int {
	M := make(map[byte]int)
	for i := 0; i < len(J); i++ {
		M[J[i]] = 1
	}

	c := 0
	for i := 0; i < len(S); i++ {
		if _, ok := M[S[i]]; ok {
			c++
		}
	}
	return c
}

func main() {
	fmt.Println("a")
}

File: J/771-jewels-and-stones/main.rs
struct Solution {}

impl Solution {
  pub fn num_jewels_in_stones(jewels: String, stones: String) -> i32 {
    let mut buf : [i32; 58] = [0; 58];
    for c in jewels.chars() {
      buf[c as usize - 'A' as usize] = 1;
    }
    stones.chars().fold(0, |acc, c| acc + buf[c as usize - 'A' as usize])
  }
}
